src/core/resource-container-helper.js

```diff
@@ -0,0 +1,146 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {rethrowAsync} from './rethrow-async';
+
+const AMP_CLASS = 'i-amphtml-element';
+const DEEP = true;
+
+const load = (element) => element.ensureLoaded();
+const pause = (element) => element.pause();
+const unmount = (element) => element.unmount();
+
+/**
+ * Ensure all elements within this container are scheduled to load.
+ *
+ * @param {!Element|!Array<!Element>} containerOrContainers
+ * @param {boolean=} includeSelf
+ */
+export function loadAll(containerOrContainers, includeSelf = true) {
+  forAllWithin(containerOrContainers, includeSelf, !DEEP, load);
+}
+
+/**
+ * Pause all elements within this container.
+ *
+ * @param {!Element|!Array<!Element>} containerOrContainers
+ * @param {boolean=} includeSelf
+ */
+export function pauseAll(containerOrContainers, includeSelf = true) {
+  forAllWithin(containerOrContainers, includeSelf, DEEP, pause);
+}
+
+/**
+ * Unmount all elements within this container.
+ *
+ * @param {!Element|!Array<!Element>} containerOrContainers
+ * @param {boolean=} includeSelf
+ */
+export function unmountAll(containerOrContainers, includeSelf = true) {
+  forAllWithin(containerOrContainers, includeSelf, DEEP, unmount);
+}
+
+/**
+ * Execute a callback for all elements within the container.
+ *
+ * @param {!Element|!Array<!Element>} containerOrContainers
+ * @param {boolean} includeSelf
+ * @param {boolean} deep
+ * @param {function(!AmpElement)} callback
+ */
+export function forAllWithin(
+  containerOrContainers,
+  includeSelf,
+  deep,
+  callback
+) {
+  if (Array.isArray(containerOrContainers)) {
+    for (let i = 0; i < containerOrContainers.length; i++) {
+      forAllWithinInternal(
+        containerOrContainers[i],
+        includeSelf,
+        deep,
+        callback
+      );
+    }
+  } else {
+    forAllWithinInternal(containerOrContainers, includeSelf, deep, callback);
+  }
+}
+
+/**
+ * Execute a callback for all elements within the container.
+ *
+ * @param {!Element} container
+ * @param {boolean} includeSelf
+ * @param {boolean} deep
+ * @param {function(!AmpElement)} callback
+ */
+function forAllWithinInternal(container, includeSelf, deep, callback) {
+  if (includeSelf && container.classList.contains(AMP_CLASS)) {
+    callCallback(callback, container);
+    if (!deep) {
+      // Also schedule amp-element that is a placeholder for the element.
+      const placeholder = container.getPlaceholder();
```

Comment:
`getPlaceholder` isn't a method on `Element`. Maybe make type `typeof HTMLElement`, so CEs can be included. We'd need to make this:
```suggestion
      const placeholder = container.getPlaceholder?.();
```

Reply:
I could do that, but to clarify. The check above checks (via class) that this is `AmpElement` which does have `getPlaceholder` method. Do you think that's not enough?